src/Http/Routing/src/Builder/EndpointRouteBuilderExtensions.cs

```diff
@@ -187,5 +187,108 @@ public static class EndpointRouteBuilderExtensions
 
             return dataSource.AddEndpointBuilder(builder);
         }
+
+        /// <summary>
+        /// Adds a <see cref="RouteEndpoint"/> to the <see cref="IEndpointRouteBuilder"/> for the specified <paramref name="pattern"/>, and
+        /// <paramref name="action"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="IEndpointRouteBuilder"/>.</param>
+        /// <param name="pattern">The route pattern.</param>
+        /// <param name="action">The action to apply to the <see cref="IApplicationBuilder"/>.</param>
+        /// <returns>A <see cref="IEndpointConventionBuilder"/> that can be used to further customize the endpoint.</returns>
+        public static IEndpointConventionBuilder MapMiddleware(
```

Comment:
Middleware is not the best term here. How well would these work if they were just a overloads of Map? Would the arguments conflict too much?

Reply:
How about MapMiddlewareEndpoint? It will be similar to MapMiddlewareRoute.